discovery-provider/src/tasks/index_rewards_manager.py

```diff
@@ -209,11 +222,34 @@ def fetch_and_parse_sol_rewards_transfer_instruction(
             return tx_metadata
 
         challenge_id, specifier = transfer_instruction
+        receiver_index = instruction["accounts"][TRANSFER_RECEIVER_ACCOUNT_INDEX]
+        pre_balance_dict = next(
+            (
+                balance
+                for balance in meta["preTokenBalances"]
+                if balance["accountIndex"] == receiver_index
+            ),
+            None,
+        )
+        post_balance_dict = next(
+            (
+                balance
+                for balance in meta["postTokenBalances"]
+                if balance["accountIndex"] == receiver_index
+            ),
+            None,
+        )
+        if pre_balance_dict is None or post_balance_dict is None:
+            raise Exception("Reward recipient balance missing!")
+        pre_balance = int(pre_balance_dict["uiTokenAmount"]["amount"])
+        post_balance = int(post_balance_dict["uiTokenAmount"]["amount"])
```

Comment:
Wondering if we could pull out a function that extracts the pre + post balances from a transaction, since we're doing this same logic in 3 places now?